datastore/usermanager.go

```diff
@@ -0,0 +1,211 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datastore
+
+import (
+	"database/sql"
+	"log"
+)
+
+const createUserTableSQL = `
+	CREATE TABLE IF NOT EXISTS userinfo (
+		id               serial primary key not null,
+		username         varchar(200) not null unique,
+		openid_identity  text not null,
+		name             varchar(200),
+		email            varchar(255) not null,
+		role             int not null
```

Comment:
I think that 'role' is a reserved word - Postgres has stolen all the good names!